builtin/k8s/releaser.go

```diff
@@ -65,6 +67,13 @@ func (r *Releaser) resourceManager(log hclog.Logger, dcr *component.DeclaredReso
 			resource.WithDestroy(r.resourceServiceDestroy),
 			resource.WithStatus(r.resourceServiceStatus),
 		)),
+		resource.WithResource(resource.NewResource(
```

Comment:
Just a note - I feel like it might make sense make resource manager aware that some resources are optional. Right now, it calls create, destroy, and status on every resource, and we just return nil quickly if it isn't enabled. It might be nice to add `resource.withOptional(func)`, rm would call that before create, and then not call any other resource function if the resource isn't enabled.

Reply:
That would be nice. We'd need that function to basically asses the config passed in to figure out whether or not it should be called or maybe just accept some kind of anonymous function so the caller can dictate when the optional resource should be used.

I wouldn't gate this PR on that feature existing, but maybe that's something we could tackle soon and remove the early return check at the beginning of some of these functions once it's in place. I am sure larger services like ECS will have optional resources too (if not already)!